core/json/src/test/scala/net/liftweb/json/ExtractionBugs.scala

```diff
@@ -96,4 +96,30 @@ object ExtractionBugs extends Specification {
       case e => e.getMessage mustEqual "Do not know how to convert JString(hi) into int"
     }
   }
+
+  "deserialize list of homogonous tuples" in {
+    implicit val formats = DefaultFormats
+
+    case class Holder(items: List[(String, String)])
+
+    val holder = Holder(List(("string", "string")))
+    val serialized = compactRender(Extraction.decompose(holder))
+
+    val deserialized = parse(serialized).extract[Holder]
+    deserialized must_== holder
+  }
+
+  "deserialize list of heterogenous tuples" in {
+    implicit val formats = DefaultFormats
+
+    // MSF: This currently doesn't work with scala primitives?! The type arguments appear as
+    // java.lang.Object instead of scala.Int. :/
```

Comment:
> Is this still true?

I'm not sure. My memory seems to think "no" is the answer there, but that could be a 2.12 development which would still make it true for 2.11.

> And if it is still true, is that something that was already missing in the standard object serialization of tuples, or not so much?

If it is still true, it would be reflective of an issue that has always been latent in scalap, I think. Whether or not we triggered that error condition in the old implementation: I don't know.

Let me do some more research on the above.